lib/cisco_node_utils/environment.rb

```diff
@@ -0,0 +1,93 @@
+# March 2016, Glenn F. Matthews
+#
+# Copyright (c) 2016 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'yaml'
+require_relative 'logger'
+
+module Cisco
+  # Class representing the configuration environment
+  class Environment
+    @environments = {}
+    @default_environment_name = 'default'
+
+    class << self
```

Comment:
This is how to create a *class* attr_accessor (`Cisco::Environment.default_environment_name`) instead of an instance attr_accessor (`foo = Cisco::Environment.new ; foo.default_environment_name`)